src/extensions/common/getRecordingInfo.ts

```diff
@@ -0,0 +1,81 @@
+import { useContext, useRef, useState } from "react";
+import { Recording, RecordingInfo, Sorting, SortingInfo } from "../extensionInterface";
+import { HitherContext, HitherInterface, HitherJob, useHitherJob } from "./hither";
+
+export const getRecordingInfo = async (a: {recordingObject: any, hither: HitherInterface}): Promise<RecordingInfo> => {
+    const recordingInfoJob = a.hither.createHitherJob(
+        'createjob_get_recording_info',
+        { recording_object: a.recordingObject },
+        {
+            useClientCache: true
+        }
+    )
+    const info = await recordingInfoJob.wait();
+    return info as RecordingInfo;
+}
+
+export const useRecordingInfo = (recordingObject: any): RecordingInfo | undefined => {
+    const {result: recordingInfo} = useHitherJob<RecordingInfo>(
+        recordingObject ? 'createjob_get_recording_info': '',
+        {recording_object: recordingObject},
+        {useClientCache: true}
+    )
+    return recordingInfo
+}
+
+export const useRecordingInfos = (recordings: Recording[]): {[key: string]: RecordingInfo} => {
+    const hither = useContext(HitherContext)
```

Comment:
Similar to useRecordingInfo, except retrieves info for a list of recordings. A nice feature of this function is that it will return partially complete results... so if we have a table of 100 recordings, we will see the data getting filled in as it becomes available